pkg/credentials/credentials.go

```diff
@@ -598,5 +599,61 @@ func vmwareCloudDirectorValidationFunc(creds map[string]string) error {
 		}
 	}
 
+	var (
+		apiToken bool
+		password bool
+		username bool
+	)
+
+	if v, ok := creds[VMwareCloudDirectorAPIToken]; ok && len(v) != 0 {
+		apiToken = true
+	}
+
+	if v, ok := creds[VMwareCloudDirectorUsername]; ok && len(v) != 0 {
+		username = true
+	}
+
+	if v, ok := creds[VMwareCloudDirectorPassword]; ok && len(v) != 0 {
+		password = true
+	}
+
+	userCreds := username || password
+
+	// API token and user credentials are mutually exclusive.
+	if userCreds && apiToken {
+		return fail.CredentialsError{
+			Op:       "validating",
+			Provider: "VMware Cloud Director",
+			Err: errors.Errorf(
+				"both api token (%s) and user credentials (%s %s) found",
+				VMwareCloudDirectorAPIToken,
+				VMwareCloudDirectorUsername,
+				VMwareCloudDirectorPassword,
+			),
+		}
+	}
+
+	// No valid credentials found.
+	if !userCreds && !apiToken {
+		return fail.CredentialsError{
+			Op:       "validating",
+			Provider: "VMware Cloud Director",
+			Err:      errors.New("no valid credentials (either api token or user) found"),
+		}
+	}
+
+	// Username and password are required when using default credentials i.e. username and password.
+	if !(username && password) {
+		return fail.CredentialsError{
+			Op:       "validating",
+			Provider: "VMware Cloud Director",
+			Err: errors.Errorf(
+				"key %v or %v is required but isn't present",
```

Comment:
Nit:
```suggestion
				"key %v and %v are required but not present",
```

Reply:
I think the original one is fine.

Reply:
The original one implies that you need only one (`or`), but my understanding is that we need both, right?

Reply:
I have committed whatever your suggestion was, should be good now.